firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -142,22 +142,25 @@ public InstallationResponse createFirebaseInstallation(
         httpsURLConnection.addRequestProperty(X_ANDROID_IID_MIGRATION_KEY, iidToken);
       }
 
-      GZIPOutputStream gzipOutputStream =
-          new GZIPOutputStream(httpsURLConnection.getOutputStream());
+      GZIPOutputStream gzipOutputStream = null;
       try {
+        gzipOutputStream = new GZIPOutputStream(httpsURLConnection.getOutputStream());
```

Comment:
This does not seem to throw a JSONException...
Why did you pull it in the try-catch block?
You can still keep the finally block without moving it in, I think.

Reply:
You are right. Good catch :)